ospackage/recipes/BuildCommand.go

```diff
@@ -0,0 +1,100 @@
+)
+
+func BuildCommand(step *osconfigpb.SoftwareRecipe_Step, artifacts map[string]string) ([]string, error) {
+	switch v := step.Step.(type) {
+	case *osconfigpb.SoftwareRecipe_Step_FileCopy:
+		return StepFileCopy(v, artifacts)
+	case *osconfigpb.SoftwareRecipe_Step_ArchiveExtraction:
+		return StepArchiveExtraction(v, artifacts)
+	case *osconfigpb.SoftwareRecipe_Step_MsiInstallation:
+		return StepMsiInstallation(v, artifacts)
+	case *osconfigpb.SoftwareRecipe_Step_DpkgInstallation:
+		return StepDpkgInstallation(v, artifacts)
+	case *osconfigpb.SoftwareRecipe_Step_RpmInstallation:
+		return StepRpmInstallation(v, artifacts)
+	case *osconfigpb.SoftwareRecipe_Step_FileExec:
+		return StepFileExec(v, artifacts)
+	case *osconfigpb.SoftwareRecipe_Step_ScriptRun:
+		return StepScriptRun(v, artifacts)
+	default:
+		return nil, fmt.Errorf("I don't know about step type %T!\n", v)
+	}
+}
+
+func StepFileCopy(step *osconfigpb.SoftwareRecipe_Step_FileCopy, artifacts map[string]string) ([]string, error) {
+	fmt.Println("StepFileCopy")
+	return nil, nil
+}
+
+func StepArchiveExtraction(step *osconfigpb.SoftwareRecipe_Step_ArchiveExtraction, artifacts map[string]string) ([]string, error) {
+	fmt.Println("StepArchiveExtraction")
+	return nil, nil
+}
+
+func StepMsiInstallation(step *osconfigpb.SoftwareRecipe_Step_MsiInstallation, artifacts map[string]string) ([]string, error) {
+	fmt.Println("StepMsiInstallation")
+	return nil, nil
+}
+
+func StepDpkgInstallation(step *osconfigpb.SoftwareRecipe_Step_DpkgInstallation, artifacts map[string]string) ([]string, error) {
+	fmt.Println("StepDpkgInstallation")
+	return nil, nil
+}
+
+func StepRpmInstallation(step *osconfigpb.SoftwareRecipe_Step_RpmInstallation, artifacts map[string]string) ([]string, error) {
+	fmt.Println("StepRpmInstallation")
+	return nil, nil
+}
+
+func StepFileExec(step *osconfigpb.SoftwareRecipe_Step_FileExec, artifacts map[string]string) ([]string, error) {
+	var path string
+	switch v := step.FileExec.LocationType.(type) {
+	case *osconfigpb.SoftwareRecipe_Step_ExecFile_LocalPath:
+		path = v.LocalPath
+	case *osconfigpb.SoftwareRecipe_Step_ExecFile_ArtifactId:
+		var ok bool
+		path, ok = artifacts[v.ArtifactId]
+		if !ok {
+			return nil, fmt.Errorf("%q not found in artifact map", v.ArtifactId)
+		}
+	default:
+		return nil, fmt.Errorf("Can't determine location type")
+	}
+
+	res := []string{path}
+	res = append(res, step.FileExec.Args...)
+	return res, nil
+}
+
+func StepScriptRun(step *osconfigpb.SoftwareRecipe_Step_ScriptRun, artifacts map[string]string) ([]string, error) {
+	// TODO: should be putting this in stepN_type/ dir, but that needs me to know the dir way in advance..
+	// actually this is an artifact. we should have made them referenced as artifacts.. we'll have to repeat file creation logic here.
+	f, err := os.Create("/tmp/scriptrun")
```

Comment:
ioutil.TempDir()

Reply:
this is a placeholder as mentioned in the todo..we don't want real 'temp' dirs.